vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/http/HttpRedirectsProvider.java

```diff
@@ -32,20 +34,20 @@ public class HttpRedirectsProvider
 
     /**
      * Executes HEAD request to get redirects.
-     * Throws IllegalStateException in case of status code outside of "200-207"
+     * Throws HttpResponseException in case of status code outside of "200-207"
      * @param from URI to issue HEAD request
      * @return List of redirects. Empty list if there are no redirects.
      */
-    public List<URI> getRedirects(URI from)
+    @SuppressWarnings("AvoidHidingCauseException")
+    public List<URI> getRedirects(URI from) throws IOException
     {
+        HttpClientContext httpContext = HttpClientContext.create();
+        HttpResponse response;
         try
         {
-            HttpClientContext httpContext = HttpClientContext.create();
-            httpClient.doHttpHead(from, httpContext).verifyStatusCodeInRange(HttpStatus.SC_OK,
-                    HttpStatus.SC_MULTI_STATUS);
-            return httpContext.getRedirectLocations().getAll();
+            response = httpClient.doHttpHead(from, httpContext);
         }
-        catch (IOException | ExternalServiceException e)
+        catch (IOException e)
         {
             String exceptionMsg;
             if (e.getCause() instanceof CircularRedirectException)
```

Comment:
are you sure that CircularRedirectException can be cause of IOException?

Reply:
> are you sure that CircularRedirectException can be cause of IOException?

As `HttpException` it should set as cause of `ClientProtocolException` (`IOException`)
https://github.com/apache/httpcomponents-client/blob/master/httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/CloseableHttpClient.java#L261

Reply:
In fact, the exception should be wrapped even here
https://github.com/apache/httpcomponents-client/blob/master/httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/InternalHttpClient.java#L173